applications/minotari_node/src/stratum/tari_sha3x_stratum_handler.rs

```diff
@@ -0,0 +1,356 @@
+                    job_id: job.job_id.to_string(),
+                    nonce,
+                    pow_algo: job.pow_algo.clone(),
+                    original_mining_hash: job.original_mining_hash.clone(),
+                    cuckaroo_nonces: cuckaroo_nonces.unwrap_or_default(),
+                },
+                tx,
+            ))
+            .await?;
+        let res = rx.await?;
+        match res {
+            Ok(_) => {
+                // Handle successful submission
+                Ok(SubmitResponse {
+                    id,
+                    result: true, // Placeholder for actual result
+                })
+            },
+            Err(e) => {
+                return Err(anyhow::anyhow!("Failed to submit job: {}", e));
+            },
+        }
+        // Handle submit request
+    }
+
+    async fn subscribe(&self, _id: String, _agent: String) -> anyhow::Result<SubscribeResponse> {
+        let mut r = OsRng;
+        let subscription_id = hex::encode(r.next_u64().to_le_bytes());
+
+        let nonce = rand::random::<u16>();
+
+        let xn = hex::encode(&nonce.to_le_bytes());
+
+        Ok(SubscribeResponse {
+            subscription_id: subscription_id.clone(),
+            nonce_hex: xn.clone(),
+            nonce,
+        })
+    }
+
+    async fn authorize(
+        &self,
+        _id: String,
+        main_algo: String,
+        login: String,
+        _worker_name: Option<String>,
+        _pass: String,
+        nonce: Option<u16>,
+    ) -> anyhow::Result<AuthorizeResponse> {
+        let algo = main_algo.parse()?;
+        let address = TariAddress::from_str(&login)?;
+
+        let job_record = create_job_from_blob(&self.block_template_repository, algo, address, nonce).await?;
+        self.job_repository.insert_job(job_record.clone()).await?;
+
+        Ok(AuthorizeResponse {
+            blob: hex::encode(job_record.blob.clone()),
+            extra_nonce_hex: job_record.xn.to_string(),
+            height: job_record.height,
+        })
+    }
+
+    async fn check_notify_needed(&self, last_job_id: String) -> Result<Option<NotifyResponse>, anyhow::Error> {
+        let last_job = self.job_repository.get_job(last_job_id.clone()).await?;
+
+        if last_job.is_none() {
+            return Err(anyhow::anyhow!("Job with id {} not found", last_job_id));
+        }
+
+        let last_job = last_job.unwrap();
+
+        let (best_height, best_block_header) = self.block_template_repository.get_tip().await?;
+
+        if last_job.height == best_height.saturating_add(1) && last_job.prev_block_hash == best_block_header {
+            // No need to notify, the job is still valid
+            return Ok(None);
+        }
+
+        // Else, let's make a new job.
+        let job_record = create_job_from_blob(
+            &self.block_template_repository,
+            last_job.pow_algo,
+            last_job.miner_address.clone(),
+            Some(last_job.xn.clone()),
+        )
```

Comment:
`Some(last_job.xn.clone())` is not used in `create_job_from_blob`
```suggestion
        let job_record = create_job_from_blob(
            &self.block_template_repository,
            last_job.pow_algo,
            last_job.miner_address.clone(),
            Some(last_job.xn.clone()),
        )
```